### PR TITLE
Accept phone number on create project

### PR DESCRIPTION
**Pull request by IniZio**
ref #3873

### CODE REVIEW

File: pkg/portal/graphql/app_mutation.go [hunk 1]
```diff
@@ -385,6 +390,7 @@ var _ = registerMutationField(
 		Resolve: func(p graphql.ResolveParams) (interface{}, error) {
 			input := p.Args["input"].(map[string]interface{})
 			appID := input["id"].(string)
+			phoneNumber := input["phoneNumber"].(string)
```
Comment (louischan-oursky) [inline pkg/portal/graphql/app_mutation.go:393]: I set the custom attribute to be of type Phone Number. This type will ensure the phone number is in E.164 format. But the phone number actually submitted to Admin API is just the national number (that is, `98765432` instead of `+85298765432`).

File: pkg/portal/model/onboard.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package model
+
+type OnboardEntry struct {
+	PhoneNumber string `json:"onboard_phone_number"`
```
Comment (louischan-oursky) [inline pkg/portal/model/onboard.go:4]: Rename the json key to `onboarding_survey_phone_number`.

File: pkg/portal/graphql/app_mutation.go (hunk 1 above)
Comment (IniZio) [reply pkg/portal/graphql/app_mutation.go:393]: Updated to include dial code :pray:

File: portal/src/PhoneTextField.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ export interface PhoneTextFieldProps {
   pinnedList?: string[];
   allowlist?: string[];
   inputValue: string;
-  onChange: (validValue: string, inputValue: string) => void;
+  onChange: (validValue: string, inputValue: string, dialCode?: string) => void;
```
Comment (louischan-oursky) [inline portal/src/PhoneTextField.tsx:15]: No need to add dialCode. `validValue` is already the E.164 value we want.

Revert the changes in this file.

File: portal/src/graphql/portal/CreateProjectScreen.tsx [hunk 4]
```diff
@@ -106,10 +108,8 @@ function CreateProjectScreenContent(props: CreateProjectScreenContentProps) {
   );
 
   const onChangePhoneNumber = useCallback(
-    (_e, newValue) => {
-      if (newValue != null) {
-        setState((prev) => ({ ...prev, phoneNumber: newValue }));
-      }
+    (_e, phoneNumber: string, dialCode?: string) => {
```
Comment (louischan-oursky) [inline portal/src/graphql/portal/CreateProjectScreen.tsx:111]: The `_e` is actually the E.164 phone number we need. Refer to PhoneScreen to see PhoneScreen stores both values for different purpose.
